src/main/java/org/jabref/logic/layout/format/FileLinkPreferences.java

```diff
@@ -2,18 +2,19 @@
 
 import java.nio.file.Path;
 import java.util.List;
+import java.util.Optional;
 
 public class FileLinkPreferences {
 
-    private final String mainFileDirectory;
+    private final Optional<Path> mainFileDirectory;
     private final List<Path> fileDirForDatabase;
 
-    public FileLinkPreferences(String mainFileDirectory, List<Path> fileDirForDatabase) {
+    public FileLinkPreferences(Optional<Path> mainFileDirectory, List<Path> fileDirForDatabase) {
```

Comment:
I really don't like the optional as a method argument here. Does not seem to provide any advantage of just returning Optional.offNullable in getMainFileDirecty but bad readability, and construction of a superfluous optional object...

Reply:
This is mostly to support `new FileLinkPreferences(getFilePreferences().getFileDirectory(),...);` where getFileDirectory returns an optional.

Reply:
Just thinking loud:
- Should mainFileDirectory ever be empty? Do we have a default value for this -> .orElse()? Reason: Have guaranteed non-null objects to avoid later checking for nulls or empty.
- Maybe overload the constructor to avoid "Optional.empty()"

Reply:
Olly's idea was to add a default for `mainFileDirectory`. But in case we don't find a nice solution for getting the user doc folder, the idea of this PR is to make it clear that the main file directory might not be present.

Reply:
next quick thought without deeper thinking 😅 : why is the mainFileDirectory not just the first Value of fileDirForDatabase?

Reply:
> Olly's idea was to add a default for `mainFileDirectory`. 

Yes!

> But in case we don't find a nice solution for getting the user doc folder, 

AWT is the currently best solution - AWT is still bundled with Java.